### PR TITLE
Bug/skale 3165 fix wait for function

### PR DESCRIPTION
**Pull request by olehnikolaiev**
low risk
get broadcast data from events if sgx crashes
fix `wait_for_fail` function

### CODE REVIEW

File: tools/bls/dkg_client.py [hunk 1]
```diff
@@ -306,6 +306,9 @@ def send_complaint(self, to_node):
             )
             logger.info(f'sChain: {self.schain_name}. '
                         f'{self.node_id_dkg} node sent a complaint on {to_node} node')
+            if len(tx_res.receipt['logs']) != 5:
```
Comment (DmytroNazarenko) [inline tools/bls/dkg_client.py]: Better move 5 to constants

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: tools/bls/dkg_client.py (hunk 1 above)
Comment (dmytrotkk) [reply tools/bls/dkg_client.py]: agree

File: tools/bls/dkg_client.py [hunk 2]
```diff
@@ -195,10 +204,21 @@ def broadcast(self, poly_name):
...
+                self.incoming_verification_vector[from_node] = convert_hex_to_g2_array(
+                    broadcasted_data[0]
+                )
+                self.incoming_secret_key_contribution[from_node] = broadcasted_data[1][
+                    192 * self.node_id_dkg: 192 * (self.node_id_dkg + 1)
+                ]
+            return
+
         self.incoming_verification_vector[from_node] = broadcasted_data[0]
         self.incoming_secret_key_contribution[from_node] = broadcasted_data[1][
```
Comment (dmytrotkk) [inline tools/bls/dkg_client.py]: looks like this calculation is identical to the line 212/213 and looks pretty complicated.
mb it makes sense to move them to the separate function?

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

File: tools/bls/dkg_client.py (hunk 2 above)
Comment (olehnikolaiev) [reply tools/bls/dkg_client.py]: fixed

**Review by DmytroNazarenko — DISMISSED** [review]
Great!

**Review by DmytroNazarenko — APPROVED** [review]

**Review by dmytrotkk — APPROVED** [review]
great

**Pull request merged**